code/player/Inventory.cs

```diff
@@ -21,20 +21,13 @@ public Inventory(TTTPlayer player) : base(player)
 
         public override void DeleteContents()
         {
-            foreach (Entity entity in List)
-            {
-                if (entity is IItem item)
-                {
-                    item.Remove();
-                }
-            }
```

Comment:
This is needed to call the `OnRemove` method for weapons as well

Reply:
ah

Reply:
Haha yea, such confusing, but to keep it in line with the other Items.